app/src/main/kotlin/com/passwordstore/android/ui/activity/SplashActivity.kt

```diff
@@ -4,18 +4,35 @@
  */
 package com.passwordstore.android.ui.activity
 
+import android.content.Intent
+import android.content.SharedPreferences
 import android.os.Bundle
 import android.view.ViewGroup
+import androidx.preference.PreferenceManager
 import com.passwordstore.android.databinding.ActivitySplashBinding
 import com.passwordstore.android.ui.EdgeToEdge
+import com.passwordstore.android.utils.Constants.IS_FIRST_RUN
 
 class SplashActivity : BaseActivity() {
+
     private lateinit var binding: ActivitySplashBinding
+    private lateinit var prefs: SharedPreferences
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        prefs = PreferenceManager.getDefaultSharedPreferences(this)
```

Comment:
Inject this through DI. I kinda wanna explore Dagger now that I migrated Viscerion (https://github.com/msfjarvis/viscerion/pull/214) and have a better understanding of everything.

Reply:
Will do once we implement DI, Dagger looks cool imo, we can try that.